### ISSUE TITLE
nodal force distribution

### ISSUE THREAD

**Issue by hex41434** [sfepy/sfepy#713]
Hi everyone! I am a little familiar with FEM and have recently simulated a simple model, in which a point force is applied on a metal part and the metal part would deform. for a part of my research, I need to export a matrix from the FEM module, which might sound a bit strange to you!!

As far as I understood: in FEM we have: F=KD (Force Matrix, Stiffness and Displacement)
F is a matrix that represents Force for all nodes of the fem mesh. ( I think it is called nodal force distribution) this matrix should be calculated in the algorithm, and later, the K matrix will be made, to produce output D (displacement of nodes).

for some reason, I need the force vector which is applied on all nodes. (not the reaction force/ nor stress/ nor strain–> these are the outputs of FEM which I do not need!)

so, my question: is this **sfepy.terms.terms_point.ConcentratedPointLoadTerm** , what I am looking for? and if not, does anybody have an idea to extract this matrix from source code? 

thanks in advance

**Comment by fisale** [comment]
Hi,

please have a look to (reaction forces in the nodes of a given region):
https://mail.python.org/archives/list/sfepy@python.org/message/O4XAUHW2S4FAXI4AX7TUCLDCKBTKF3NF/

Perhaps this might help you.

**Comment by rc** [comment]
Please clarify: you need the components of the assembled linear system, that is the vector F and the matrix K, that respect the boundary conditions?

@fisale's comment (thanks!)  shows how to get the reaction forces in nodes fixed by Dirichlet boundary conditions.

**Comment by hex41434** [comment]
> Please clarify: you need the components of the assembled linear system, that is the vector F and the matrix K, that respect the boundary conditions?
> 
> @fisale's comment (thanks!) shows how to get the reaction forces in nodes fixed by Dirichlet boundary conditions.

Yes, exactly! I need the elements of Force Vector. is this easy to extract it from sfepy?

**Comment by rc** [comment]
To get the vector F you would need to use the imperative (interactive) way of problem description. There is no example that shows that, but basically you can follow any example that has '_interactive' in its name, e.g. [1] up to `pb.solve()` call, and then:
```
pb.time_update()
state = pb.create_state()
state.apply_ebc()
R = pb.equations.eval_residuals(state())
```
Note that this computes R = F - K D0, where D0 are the initial displacements (usually zeros + applied Dirichlet BCs). See [2] for explanation.

[1] https://sfepy.org/doc-devel/examples/linear_elasticity-linear_elastic_interactive.html
[2] https://sfepy.org/doc-devel/ebcs_implementation.html
